charts/orchestrator/templates/tekton-tasks.yaml

```diff
@@ -0,0 +1,265 @@
+metadata:
+  name: flattener
+  namespace: {{ .Values.orchestrator.namespace }}
+spec:
+  workspaces:
+    - name: workflow-source
+  params:
+    - name: workflowId
+      description: The workflow ID from the repository
+      type: string
+    - name: convertToFlat
+      description: Whether conversion to flat layout is needed or it's already flattened
+      type: string
+      default: "true"
+  steps:
+    - name: flatten
+      image: registry.access.redhat.com/ubi9-minimal
+      workingDir: $(workspaces.workflow-source.path)
+      script: |
+        echo "--pwd"
+        pwd
+        echo "--ls"
+        ls
+        echo "--ls /workspace/*"
+        ls /workspace/*
+        ROOT=/workspace/workflow
+        TARGET=flat
+        mkdir -p flat
+        if [ "$(params.convertToFlat)" == "true" ]; then
+          if [ -d "workflow/$(params.workflowId)" ]; then
+            cp -r workflow/$(params.workflowId)/src/main/resources flat/$(params.workflowId)
+          else
+            cp -r workflow/src/main/resources flat/$(params.workflowId)
+          fi
+        else
+          cp -r workflow/$(params.workflowId) flat/$(params.workflowId)
+        fi
+
+        curl -L https://raw.githubusercontent.com/parodos-dev/serverless-workflows/main/pipeline/workflow-builder.Dockerfile -o flat/workflow-builder.Dockerfile
+
+        echo "--ls /workspace"
+        ls /workspace
+        echo "--ls /workspace/*"
+        ls /workspace/*
+        echo "--ls -l flat"
+        ls -l flat
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: build-manifests
+  namespace: {{ .Values.orchestrator.namespace }}
+spec:
+  workspaces:
+    - name: workflow-source
+  params:
+    - name: workflowId
+      description: The workflow ID from the repository
+      type: string
+  steps:
+    - name: build-manifests
+      image: registry.access.redhat.com/ubi9-minimal
+      workingDir: $(workspaces.workflow-source.path)/flat/$(params.workflowId)
+      script: |
+        echo "--pwd"
+        pwd
+        echo "--ls"
+        ls
+        curl -L https://github.com/rgolangh/kie-tools/releases/download/0.0.2/kn-workflow-linux-amd64 -o kn-workflow
+        chmod +x kn-workflow
+        echo "Installed kn-workflow"
+        ./kn-workflow gen-manifest --namespace ""
+        echo "--ls manifests"
+        ls manifests
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: build-config
+  namespace: {{ .Values.orchestrator.namespace }}
+spec:
+  workspaces:
+    - name: workflow-source
+    - name: workflow-config
+  params:
+    - name: workflowId
+      description: The workflow ID from the repository
+      type: string
+    - name: imageTag
+      type: string
+  steps:
+    - name: build-config
+      image: registry.access.redhat.com/ubi9-minimal
+      workingDir: $(workspaces.workflow-config.path)/workflow-config
+      script: |
+        echo "--pwd"
+        pwd
+        echo "--ls"
+        ls
+        echo "--ls ../.."
+        ls ../..
```

Comment:
why is it needed? For debug purposes?

Reply:
Yes, I had an issue to remove them in the original repo: https://github.com/parodos-dev/workflows-tekton-pipeline/issues/6
I can do that in the next commit